intlist.c

```diff
@@ -0,0 +1,269 @@
+         temp2 = temp->link;
+         free (temp);
+         temp = NULL;
+         temp = temp2;
+      }
+      printf ("\nList deleted... \n");
+      *head = NULL;
+      return;
+   }
+}
+
+//Prints the number of elements in the list
+void Count (struct node** head) {
+   struct node* store = *head;
+   int count = 0;
+   while (store != NULL) {
+      store = store->link;
+      count++;
+   }
+   printf ("\nThe number of elements in the list is %d\n", count);
+}
+
+//Returns the number of elements in the list
+int CountNum (struct node** head) {
+   struct node* store = *head;
+   int count = 0;
+   while (store != NULL) {
+      store = store->link;
+      count++;
+   }
+   return count;
+}
+
+//Removes the element at a zero based index position from the list
+void RemoveAt (struct node** head, int n) {
+   int count = 0;
+   int lengthOfList = CountNum (head);
+   //Comparision variable that stores the number of times the loop is needed to iterate from the head 
+   // to the required zero based index position  
+   int cmp = lengthOfList - n - 1;
+   if (*head == NULL) {
+      printf ("\nThe list is empty \n");
+      return;
+   }
+   if (n < 0) {
+      printf ("\nInvalid index \n");
+   }
+   if (n > (lengthOfList - 1)) {
+      printf ("\nThe index exceeds the length of the list (zero based) \n");
+      return;
+   }
+   //To remove the first element of the list
+   if (n == 0) {
+      struct node* temp = *head;
+      struct node* prev = *head;
+    //To remove the zero index element when it's the only element in the list
+      if (((*head)->link) == NULL) {
+         printf ("%d is removed from the list\n", temp->value);
+         free (temp);
+         *head = NULL;
+         return;
+      }
+    //To remove the first element when it's not the only element in the list
+      while ((temp->link) != NULL) {
+         prev = temp;
+         temp = temp->link;
+      }
+      printf ("\n%d is removed from the list\n", temp->value);
+      prev->link = NULL;
+      free (temp);
+      PrintFunc (*head);
+      return;
+   }
+   //To remove the last element of the list 
+   if (n == (lengthOfList - 1)) {
+      struct node* temp;
+      temp = *head;
+      printf ("\n%d is removed from the list\n", temp->value);
+      *head = (*head)->link;
+      PrintFunc (*head);
+      free (temp);
+      temp = NULL;
+      return;
+   }
+   struct node* prev = *head;
+   struct node* temp = *head;
+   while ((temp != NULL) && (count < (cmp))) {
+      prev = temp;
+      temp = temp->link;
+      count++;
+
+   };
+   //tempNext points to the next node from the element to be removed 
+   struct node* tempNext = NULL;
+   if (n == 0) {
+      tempNext = NULL;
+   } else {
+      tempNext = temp->link;
+   }
+   prev->link = tempNext;
+   printf ("\n%d is removed from the list\n", temp->value);
+   free (temp);
+   PrintFunc (*head);
+}
+
+//Gets the element from the given index position

```

Comment:
How is the calling function supposed to actually "get" the value of the element? You are just printing it.